### PR TITLE
Update gimli

### PR DESCRIPTION
**Pull request by philipc**
These are the changes to go with https://github.com/gimli-rs/gimli/pull/382.

Let me know if there are any changes you disagree with, or if you have any ideas for further improvements.

### CODE REVIEW

File: src/debuginfo.rs [hunk 1]
```diff
@@ -51,29 +50,36 @@ fn line_program_add_file(line_program: &mut LineProgram, line_strings: &mut Line
...
-    line_programs: LineProgramTable,
-
-    // Side tables
-    strings: StringTable,
-    line_strings: LineStringTable,
-
-    // Global ids
-    unit_id: UnitId,
-    global_line_program: LineProgramId,
+    dwarf: DwarfUnit,
```
Comment (bjorn3) [inline src/debuginfo.rs]: :+1:

Could you please remove the two comments above?

File: src/debuginfo.rs [hunk 2]
```diff
@@ -183,73 +175,37 @@ impl<'a, 'tcx: 'a> DebugContext<'tcx> {
...
-        let debug_ranges_not_empty = !debug_ranges.0.writer.slice().is_empty();
-        if debug_ranges_not_empty {
-            decl_section!(DebugRanges = debug_ranges);
-        }
-
-        let debug_rnglists_not_empty = !debug_rnglists.0.writer.slice().is_empty();
-        if debug_rnglists_not_empty {
-            decl_section!(DebugRngLists = debug_rnglists);
-        }
+        let _: Result<()> = sections.for_each_mut(|id, section| {
```
Comment (bjorn3) [inline src/debuginfo.rs]: This is so much cleaner!

File: src/debuginfo.rs [hunk 3]
```diff
@@ -98,45 +106,34 @@ impl<'a, 'tcx: 'a> DebugContext<'tcx> {
...
         };
 
-        let mut strings = StringTable::default();
-        let mut line_strings = LineStringTable::default();
-
-        let mut units = UnitTable::default();
-        let mut line_programs = LineProgramTable::default();
-
-        let global_line_program = line_programs.add(LineProgram::new(
+        let line_program = LineProgram::new(
```
Comment (bjorn3) [inline src/debuginfo.rs]: Maybe add a convenience method for these values. (1, 1, -5, 14) These are the ones rustc (or LLVM itself) uses.

Comment (philipc) [reply src/debuginfo.rs]: I'm not sure what to do about these yet. Ideally I think these should be selected based on the target arch (at least the first two parameters, but maybe the others as well).

**Comment by bjorn3** [comment]
I will have to update for latest nightly. Will merge later today.

**Comment by bjorn3** [comment]
Thanks!

**Pull request merged**